src/jit/gentree.h

```diff
@@ -2107,7 +2107,7 @@ struct GenTreeLclVar: public GenTreeLclVarCommon
                             DEBUG_ARG(largeNode)),
             gtLclILoffs(ilOffs)
             {
-                assert(OperIsLocal(oper));
+                assert(OperIsLocal(oper) || OperIsLocalAddr(oper));
```

Comment:
GT_LCL_var can already be a local addr, just not at the time it is constructed (i.e. it is legal to create an lcl_var node and then change its type to a lcl_addr).